Improve using Radicle doc sections (#16)
radicle-dev/radicle-docs: docs/using-radicle/pushing-and-pulling-changes.md

## Committing and pushing code

To push code to Radicle, you first need to make sure your shell is configured correctly by following the instructions in the [Getting Started](getting-started/doc1-1.md) section.

> After the Radicle remote helper is configured, pushing to Radicle is as simple as pushing to any other platform. After changes are added and commited, they can be pushed to the Radicle remote via `git push rad`. For help using Git, please refer to [relevant documentation](https://docs.github.com/en/free-pro-team@latest/github/using-git).

Reviewer: Can we link to a resource on how to use git that is not part of the github documentation?

Author: Good point, it felt strange linking to GH to me as well. How about: https://git-scm.com/book/en/v2/Getting-Started-First-Time-Git-Setup?

Reviewer: :black_heart:

Reviewer 2: Ya fine with this. Mainly had the link as a placeholder to include some git documentation.

Author: https://github.com/radicle-dev/radicle-docs/pull/16/commits/0a69cfa041ac1b77bfde914997b6f423d2d04864

> It's important to note that once changes are pushed to the Radicle remote, they are published to the Radicle network and are available to anybody who is connected to you in the network.

Reviewer: They are only published if the Upstream client is running, we need to make that explicit. Otherwise it will be perceived as unexpected behaviour when changes are not propagated despite being pushed to the rad remote.

Reviewer 2: Cool, will edit.

> It's important to note that once changes are pushed to your Radicle remote, they are published to the Radicle network and will be publically available to anybody you are connected with. Updates are only published to the network if the Upstream client is running. 

Reviewer 3: nit/suggestion: I think "publicly" is the preferred spelling, although "publically" is a valid variant.
```suggestion
It's important to note that once changes are pushed to your Radicle remote, they are published to the Radicle network and will be publicly available to anybody you are connected with. Updates are only published to the network if the Upstream client is running. 
```

Reviewer 2: Yeah, you're right.

### Following projects

> To contribute to another user's project, you must first find it on the Radicle network. To find a project, you need its [**projectID**](understanding-radicle/glossary.md/#project-id). This can be found in the project page under the [project name](understanding-radicle/glossary.md/#project-name), or in a seed node UI.

Reviewer: We share URNs to get projects, but we only talk about the id.

Reviewer 2: Isn't the URN the projectID?

Reviewer: Nope, the urn contains it, the current format for git pased projects is: `rad:git:<project_id>`.

Reviewer 2: Gotcha. Originally, the project URN was renamed (in the app) to project handle (for simplicity & user friendliness). Then it went up in the air what this *actually* should be called. 

Seeing the format, I agree that projectID *doesn't* make sense, so I believe we should switch back to project handle.

Reviewer 2: I've proposed a resolution to this topic in [#1206](https://github.com/radicle-dev/radicle-upstream/pull/1206) and implemented changes in my latest commits.

[insert picture of projectID in seednode UI and in project page]

Once found, copy the projectID into the Upstream search bar. Following the project will replicate it to your local machine once it's found in your network of connected peers.

[insert picture of projectID & search bar]

### Cloning projects

To clone a working copy of a followed project to your computer, you can use the [**Checkout**](understanding-radicle/glossary.md/#checkout) option within the Upstream client. This will clone the project to a specified location on your local machine.

Commiting and publishing changes to this working copy will create a new [remote](understanding-radicle/glossary.md/#remote) for that project which is uniquely linked to your peerID. You can view this remote in the peer switcher.

[insert picture of peer switcher]

> Once a remote is created for a project, it will also move the project to your **My Projects** tab.

Reviewer 3: I think it's just called "Projects" in the UI, right?
```suggestion
Once a remote is created for a project, it will also move the project to your **Projects** tab.
```

## Getting changes from a project

### Adding remotes
To view and pull changes from another user's version of your project, you must first **add their remote**. To do this, you must add their **peerID** to your list of remotes.

[insert gif of adding peerID to manage remotes modal]

> Adding a user's remote is basically adding their version of your project to your machine. Once added, you'll be able to fetch changes from their version and apply them to your working copy. Read more about [how remotes work on Radicle](understanding-radicle/faq.md).

Reviewer 3: This isn't  _quite_ right. There must be a path for you to get that remote's changes. This can either be:
* A peer you're connected to has their changes and we fetch those changes from them
* You hear a gossip message about the peer you just tracked

This explanation makes it seem like this happens instantaneously.

Reviewer 2: Yes, I haven't edited this part since our conversation, but will be sure to fix it up in my next iteration.

Reviewer 2: I rewrote as such. Having trouble understanding how "hearing a gossip message" is different from fetching changes from a connected peer. How would you rewrite these sentences? 

_"Adding and following a user's remote is basically adding their version of your project to your machine. To get changes from a new remote you must fetch them from a connected peer that has replicated the remote's changes. Once fetched, you'll be able to pull changes from their version and apply them to your working copy."_

Reviewer 4: I feel like we should stay away from using "version" as it's not really a known term when it comes to git repositories. I think saying something like *... is basically adding their branches of your project to your machine* might be clearer. Otherwise we can use "fork" like github, though I think that's a bit of a mis-use of the term.

Reviewer: What about their view?

Reviewer 2: _"Adding and following a user's remote is basically adding their branches of your project to your machine. To get changes from a new remote you must fetch them from a connected peer that has replicated the remote's changes. Once fetched, you'll be able to pull changes from their version and apply them to your working copy."_ 

?

Reviewer 3: I made a suggestion here https://github.com/radicle-dev/radicle-docs/pull/16/files#r520523526. I _think_ it's related but now I'm finding it hard to tell with diffing

> Adding and following a user's remote is basically adding their version of your project to your machine. To get changes from a new remote you must fetch them from a connected peer that has replicated the remote's changes. Once fetched, you'll be able to pull changes from their version and apply them to your working copy. Read more about [how remotes work on Radicle](understanding-radicle/faq.md).

Reviewer 3: How's this? The previous version makes it sound like the user can fetch via some button or whatever, but that's not the case.

```suggestion
Adding a peer's remote means that you wish to track that peer's changes for this project. It's important to note that these changes may not instantly appear. We rely on the connectdness of the network to ensure that we hear about the changes from a peer we care about. Assuming that the network of peer's also track the peer you are looking for, you will get their changes when:
1. On startup of the application, where your peer will ask for any new changes from all the peers you track. For more information on tracking [click here](is there a tracking doc?).
2. Your peer receives a gossip message about the peer you wish to track.  For more information on gossip [click here](doc-on-gossip?).
```

Reviewer 2: Great rewrite - made some minor edits to keep language consistent. Can you review @FintanH ?

Adding a peer's remote means that you wish to track their changesets for your project. It's important to note that these changes may not instantly appear. We rely on the connectedness of the network to ensure that we hear about the changes from a peer we follow. Assuming that the network of peers also track the peer you are looking for, you will get their changes:
1. On application startup, where your peer will ask for any new changes from all the peers you follow. For more information on following, [click here](understanding-radicle/how-it-works.md/#following).
2. Whenever your peer receives a gossip message about the peer you wish to follow.  For more information on gossip, [click here](understanding-radicle/how-it-works.md/#gossip).

Reviewer 3: Looks great! :) Maybe one slight modification now that I'm reading it again:

> Assuming that the network of peers also track the peer you are looking for, you will get their changes:

changes to:

> Assuming that the network of peers includes the peer you wish to track _or_ also tracks the peer you are looking for, you will get their changes:

Reviewer 2: Cool, updated your suggestion directly with this edit. @rudolfs - if you want to commit, go ahead and then we can get this merged.

Author: https://github.com/radicle-dev/radicle-docs/pull/16/commits/92a6d1db12cefe8ccb135b074e7f5bd4a8d29fd8

You can view all remotes of a project with the command:

`git remote show`

Or, list all branches of all remotes with the command:

`git branch -a`

### Fetching changes
After a remote has been added, any changes can be fetched with:

`git fetch <remote>`

> Where `<remote>` is of the following form: `remotes/<display name>@<peerID>/<branch>` and can be looked up by inspecting the output of `git branch -a`. Remotes returned by `git branch -a` match the remotes found in the "Manage remotes" modal.

Reviewer 3: You don't need the `remotes/` prefix.

```suggestion
Where `<remote>` is of the following form: `<display name>@<peerID>/<branch>` and can be looked up by inspecting the output of `git branch -a`. Remotes returned by `git branch -a` match the remotes found in the "Manage remotes" modal.
```

`git fetch <remote>`

> Where `<remote>` is of the following form: `<display name>@<deviceID>/<branch>` and can be looked up by inspecting the output of `git branch -a`. Remotes returned by `git branch -a` match the remotes found in the "Manage remotes" modal.

Author: ```suggestion
Where `<remote>` is of the following form: `<display name>@<Device ID>/<branch>` and can be looked up by inspecting the output of `git branch -a`. Remotes returned by `git branch -a` match the remotes found in the "Manage remotes" modal.
```
